ui/segmented-bar.reel/segment.reel/segment.js

```diff
@@ -0,0 +1,93 @@
+ * A Segment composes of data that forms a SegmentedBar. A Segment
+ * is bound to an object from SegmentedBar's content array. An array will be projected
+ * as a segment. The 'contentController' (rangeController/treeController)
+ * will use the proper logic and project the array of the rangeController or
+ * flatten the treeController data. The data projected will draw itself
+ * based on a percentage of the bar's total length. The remaining space
+ * will be its own Segment in the SegmentedBar.
+ */
+exports.Segment = Component.specialize({
+    label: {value: null},
+    _value: {value: null},
+    value: {
+        get: function () {return this._value},
+        set: function (value) {
+            this._value = value;
+            this._calculateLength(value);
+            this.needsDraw = true;
+        }
+    },
+    /**
+     * Length of each segment in %
+     * calculated based on data value & SegmentedBar's maxValue
+     */
+    _length: {value: null},
+    _calculateLength: {value: function _calculateLength(value) {
+        this._length = value / this.maxValue * 100 + "%";
+    }},
+
+    enterDocument: {value: function () {
+        // set initial segment width
+        this._calculateLength(this.value);
+
+        // TODO: perf review of listening here; doesn't work in prepareForActivationEvents
+        this.addPathChangeListener("orientation", this, "handleOrientationChange");
+    }},
+
+    prepareForActivationEvents: {
+        value: function () {
+            this.element.addEventListener("click", this);
+        }
+    },
+
+    /**
+     * Handles the click of the each individual Segment
+     * Passes data to be displays in more detail elsewhere
+     */
+    handleClick: {value: function handleClick(event) {
+        // bubble a CustomEvent & stop propagating heard event
+        event.stopPropagation();
+
+        // TODO: need more data?
+        var _data = {label: this.label, value: this.value};
+        var segmentClickEvent = new CustomEvent(
+            "SegmentClick",
+            {detail: _data, bubbles: true, cancelable: true}
+        );
+        this.element.dispatchEvent(segmentClickEvent);
+    }},
+
+    handleOrientationChange: {value: function handleOrientationChange() {
+        this.needsDraw = true;
+    }},
+
+    handleSegmentResize: {value: function handleSegmentResize() {
+        // logic for hiding labels if too narrow
+        this.needsDraw = true;
+    }},
+
+    /**
+     * Assigns the length as the width or height depending on the
+     * orientation & draws it.
+     */
+    draw: {value: function () {
+        if (this.orientation === "horizontal") {
+            this.element.style.width = this._length;
+            this.element.style.height = "100%";
+            this.element.style["padding-left"] = "5px";
+        } else {
+            this.element.style.height = this._length;
+            this.element.style.width = "100%";
+            this.element.style["padding-left"] = "0px";
```

Comment:
Can be access directly by the property paddingLeft, it's faster than using the brackets.